app_test.go

```diff
@@ -226,6 +228,28 @@ func TestAppStop(t *testing.T) {
 	})
 }
 
+func TestAppRun(t *testing.T) {
+	run := func(app *App) {
+		go func() {
+			app.Run()
+		}()
+		// 100ms sleep makes sure that app is in RUNNING state
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Could you add a simple function, with start and stop hooks on a wait group. Start adds something to a wait group, stop executes `Done()` and go routine with the running app waits for the group.